components/api_server/src/model/report.py

```diff
@@ -26,25 +29,27 @@ def issue_tracker(self) -> IssueTracker:
         )
         return IssueTracker(url, issue_parameters, credentials)
 
-    def desired_response_time(self, status: str) -> int:
-        """Return the desired response time for the metric status."""
-        # Note that the frontend also has these constant, in src/defaults.js.
+    def deadline(self, status: str) -> str | None:
+        """Return the deadline for metrics or measurement entities with the given status."""
+        desired_response_time = self._desired_response_time(status)
```

Comment:
Regarding typing, it would make sense if `self._desired_response_time` would actually return a timedelta object. But I'm not sure in how many places we're relying on it being an int :)

Reply:
Yeah, I thought about that, but all users of `deadline()` convert the result to a string anyway, so that would lead to duplication.